### PR TITLE
Restyle top banner

### PR DESCRIPTION
**Pull request by EmilMalanczak**

### CODE REVIEW

File: packages/app/src/ui/atoms/sparkles/Sparkles.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
+import { cn } from '@/ui/utils/style'
+import { randomIntRange } from '@/utils/random'
+import { useCallback, useEffect, useRef, useState } from 'react'
+
+function range(start: number, end: number, step = 1) {
```
Comment (EmilMalanczak) [inline packages/app/src/ui/atoms/sparkles/Sparkles.tsx:5]: should I put this inside `src/utils`? eg. `arrays.ts` file

File: packages/app/src/ui/atoms/sparkles/Sparkles.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+export interface SparklesProps extends React.HTMLAttributes<HTMLSpanElement> {
+  color?: string
+  children: React.ReactNode
+  sizeRange?: [number, number]
+}
+
+export function Sparkles({ color = DEFAULT_COLOR, children, sizeRange = [10, 18], ...delegated }: SparklesProps) {
+  const [sparkles, setSparkles] = useState(() => range(0, 3).map(() => generateSparkle({ color, sizeRange })))
+  // const prefersReducedMotion = usePrefersReducedMotion()
```
Comment (EmilMalanczak) [inline packages/app/src/ui/atoms/sparkles/Sparkles.tsx:24]: do we care about such functionality? There is ready to copy-paste [implementation](https://www.joshwcomeau.com/snippets/react-hooks/use-prefers-reduced-motion/) but not sure if worth it

File: packages/app/src/ui/atoms/sparkles/Sparkles.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  )
+}
+
+interface UseRandomIntervalParams {
+  minDelay: number
+  maxDelay: number
+  enabled: boolean
+}
+
+function useRandomInterval(callback: () => void, { minDelay, maxDelay, enabled }: UseRandomIntervalParams) {
```
Comment (EmilMalanczak) [inline packages/app/src/ui/atoms/sparkles/Sparkles.tsx:87]: WIP - trying to reimplement this base on the `useTimestamp` hook - this makes a bit more natural feeling -> [example](https://www.joshwcomeau.com/react/animated-sparkles-in-react/#generation-and-cleanup-5)

File: packages/app/src/ui/atoms/sky-migration-banner/SkyMigrationBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { assets } from '@/ui/assets'
+import { XIcon } from 'lucide-react'
+import { Button } from '../button/Button'
+import { Link } from '../link/Link'
+import { Sparkles } from '../sparkles/Sparkles'
+
+interface SkyMigrationBanner {
```
Comment (yivlad) [inline packages/app/src/ui/atoms/sky-migration-banner/SkyMigrationBanner.tsx:7]:
```suggestion
interface SkyMigrationBannerProps {
```

**Review by yivlad — APPROVED** [review]

File: packages/app/src/ui/atoms/sky-migration-banner/SkyMigrationBanner.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  onClose: () => void
+}
+
+export const SKY_MIGRATION_BANNER_ID = 'sky-migration-banner'
```
Comment (krzkaczor) [inline packages/app/src/ui/atoms/sky-migration-banner/SkyMigrationBanner.tsx:11]: I like the idea that with generic closable banners. One comment: can we rename this component to **top banner** (not just "banner"). 

I imagine that we could have `sky-migration-banner` that looks totally differently, somewhere on the page but it's still closable.

File: packages/app/src/ui/atoms/sparkles/Sparkles.tsx (hunk 1 above)
Comment (krzkaczor) [reply packages/app/src/ui/atoms/sparkles/Sparkles.tsx:5]: yeah i think it makes sense 👍 this is useful

File: packages/app/src/ui/atoms/sparkles/Sparkles.tsx (hunk 2 above)
Comment (krzkaczor) [reply packages/app/src/ui/atoms/sparkles/Sparkles.tsx:24]: i think we can get rid of this line.

File: packages/app/src/domain/state/banners.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
+import { StateCreator } from 'zustand'
+
+import { StoreState, useStore } from '.'
+
+export interface Banners {
+  showBannerStatus: {
```
Comment (oskarvu) [inline packages/app/src/domain/state/banners.ts:6]: Maybe `bannerVisibilityStatus`?

Comment (krzkaczor) [reply packages/app/src/domain/state/banners.ts:6]: Tbf: i was also wonderingaot naming here. IMO the whole slice should be called bannersVisibility a not just banners

**Review by oskarvu — APPROVED** [review]
nice

**Pull request merged**